Row.h

```diff
@@ -0,0 +1,141 @@
+#ifndef HEADER_Row
+#define HEADER_Row
+/*
+htop - Row.h
+(C) 2004-2015 Hisham H. Muhammad
+(C) 2023 htop dev team
+Released under the GNU GPLv2+, see the COPYING file
+in the source distribution for its full text.
+*/
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <sys/types.h>
+
+#include "Object.h"
+#include "RichString.h"
+
+
+struct Table_;
+struct Machine_;
+
+/* Class representing entities (such as processes) that can be
+ * represented in a tabular form in the lower half of the htop
+ * display. */
+
+typedef struct Row_ {
+   /* Super object for emulated OOP */
+   Object super;
+
+   /* Pointer to quasi-global data */
+   const struct Machine_* host;
+
+   int id;
+   int group;
+   int parent;
+
+   /* Has no known parent */
+   bool isRoot;
+
+   /* Whether the row was tagged by the user */
+   bool tag;
+
+   /* Whether to display this row */
+   bool show;
+
+   /* Whether this row was shown last cycle */
+   bool wasShown;
+
+   /* Whether to show children of this row in tree-mode */
+   bool showChildren;
+
+   /* Whether the row was updated during the last scan */
+   bool updated;
+
+   /*
+    * Internal state for tree-mode.
+    */
+   int32_t indent;
+   unsigned int tree_depth;
+} Row;
+
+typedef unsigned int RowField;
+
+typedef Row* (*Row_New)(const struct Machine_*);
+typedef bool (*Row_Filtered)(const Row*, const struct Table_*);
+typedef bool (*Row_Obscured)(const Row*, const struct Table_*);
```

Comment:
Can we add an "is" word on these functions that return boolean?
* `Row_IsObscured`
* `Row_IsFiltered`

By the way, what are the purposes of these two new virtual functions?

Reply:
`Row_IsFiltered` is used to determine if the `Row`  matches a filter defined by the `IncSet` filter for the active `Panel`.

`Row_IsObscured` is a refactor of the check to see if a given `Row` needs to be shown or not; e.g. a LWP being hidden because of `Settings->hideUserlandThreads` being active.

Reply:
I have to update and suggest better names for these functions:

* `Row_Filtered` renamed to `Row_MatchesFilter`
* `Row_Obscured` renamed to `Row_IsVisible` and the semantics reversed.

Reply:
Yep, agreed, that's neater - I'll make those name changes, thanks @Explorer09

Reply:
This is done now in latest update, thanks @Explorer09